conjure-java-core/src/integrationInput/java/com/palantir/product/StrictFourFields.java

```diff
@@ -0,0 +1,318 @@
+            hash = 31 * hash + this.bearerTokenValue.hashCode();
+            hash = 31 * hash + this.optionalItem.hashCode();
+            hash = 31 * hash + this.mappedRids.hashCode();
+            result = hash;
+            memoizedHashCode = result;
+        }
+        return result;
+    }
+
+    @Override
+    @DoNotLog
+    public String toString() {
+        return "StrictFourFields{myList: " + myList + ", bearerTokenValue: " + bearerTokenValue + ", optionalItem: "
+                + optionalItem + ", mappedRids: " + mappedRids + '}';
+    }
+
+    private static void validateFields(
+            List<String> myList,
+            BearerToken bearerTokenValue,
+            Optional<String> optionalItem,
+            Map<ResourceIdentifier, String> mappedRids) {
+        List<String> missingFields = null;
+        missingFields = addFieldIfMissing(missingFields, myList, "myList");
+        missingFields = addFieldIfMissing(missingFields, bearerTokenValue, "bearerTokenValue");
+        missingFields = addFieldIfMissing(missingFields, optionalItem, "optionalItem");
+        missingFields = addFieldIfMissing(missingFields, mappedRids, "mappedRids");
+        if (missingFields != null) {
+            throw new SafeIllegalArgumentException(
+                    "Some required fields have not been set", SafeArg.of("missingFields", missingFields));
+        }
+    }
+
+    private static List<String> addFieldIfMissing(List<String> prev, Object fieldValue, String fieldName) {
+        List<String> missingFields = prev;
+        if (fieldValue == null) {
+            if (missingFields == null) {
+                missingFields = new ArrayList<>(4);
+            }
+            missingFields.add(fieldName);
+        }
+        return missingFields;
+    }
+
+    public static MyListStageBuilder builder() {
+        return new DefaultBuilder();
+    }
+
+    public interface MyListStageBuilder {
+        BearerTokenValueStageBuilder myList(@Nonnull Iterable<String> myList);
+
+        Builder from(StrictFourFields other);
+    }
+
+    public interface BearerTokenValueStageBuilder {
+        OptionalItemStageBuilder bearerTokenValue(@Nonnull BearerToken bearerTokenValue);
+    }
+
+    public interface OptionalItemStageBuilder {
+        MappedRidsStageBuilder optionalItem(@Nonnull Optional<String> optionalItem);
+
+        MappedRidsStageBuilder optionalItem(@Nonnull String optionalItem);
+    }
+
+    public interface MappedRidsStageBuilder {
+        Completed_StageBuilder mappedRids(@Nonnull Map<ResourceIdentifier, String> mappedRids);
+    }
+
+    public interface Completed_StageBuilder {
+        StrictFourFields build();
+
+        Completed_StageBuilder addAllMyList(@Nonnull Iterable<String> myList);
+
+        Completed_StageBuilder myList(String myList);
+
+        Completed_StageBuilder putAllMappedRids(@Nonnull Map<ResourceIdentifier, String> mappedRids);
+
+        Completed_StageBuilder mappedRids(ResourceIdentifier key, String value);
```

Comment:
Adding the auxiliary methods for collections to the final stage.

Reply:
I think the auxiliary methods on the final stage are a little awkward for a couple reason:
- In most cases (field doesn't happen to be the last one), users will have to come back to a field they've already described to use these methods
- There's no way to use an auxiliary method to specify a few entries individually instead of using the full value setter

I might propose dropping the auxiliary methods from the final stage for now. We could support them with additional complex stages, but imo that would be easier to review in a follow-up if we decide it's helpful
```

Reply:
Yeah, this was my initial thought as well, but added them to keep the definition of "strict" mean that only "all fields are set" is enforced. I think dropping the auxiliary methods for collections makes sense here. 

I think the auxiliary value setter for optionals seems ok to keep, in the stage where the optional field is set. What do you think?

Reply:
Totally agree, the convenience methods are great

Reply:
Update the PR to remove aux methods for collections.